src/app/map/search/search.component.ts

```diff
@@ -0,0 +1,156 @@
+  public _permitFilter: string = null; // temporary filters for Cancel feaure
+  public typeahead: Observable<string> = null;
+  public resultsCount = 0;
+
+  public radioSel: string;
+  public radioOptions: string[] = ['Mine Name', 'Permit Number'];
+  private mineKeys: Array<string> = [];
+  private permitKeys: Array<string> = [];
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private location: Location,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {}
+
+  // (arrow) functions to return type-ahead results
+  // ref: https://ng-bootstrap.github.io/#/components/typeahead/api
+  //
+  public mineSearch = (text$: Observable<string>) =>
+    text$
+      .debounceTime(200)
+      .distinctUntilChanged()
+      .map(term => term.length < 1 ? []
+        : this.mineKeys.filter(key => key.indexOf(this._mineFilter.toUpperCase()) > -1) // .slice(0, 10)
+      )
+
+  public permitSearch = (text$: Observable<string>) =>
+    text$
+      .debounceTime(200)
+      .distinctUntilChanged()
+      .map(term => term.length < 1 ? []
+        : this.permitKeys.filter(key => key.indexOf(this._permitFilter.toUpperCase()) > -1) // .slice(0, 10)
+      )
+
+  ngOnInit(): void {
+    this.radioSel = 'Mine Name';
+  }
+
+  public ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  public onLoadStart() { this.loading = true; }
+
+  public onLoadEnd() { this.loading = false; }
+
+  // called when apps list changes
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.projects && !changes.projects.firstChange && changes.projects.currentValue) {
+
+      this.mineKeys = _.sortedUniq(_.compact(this.projects.map(app => app.name ? app.name.toUpperCase() : null)).sort());
+      this.permitKeys = _.sortedUniq(_.compact(this.projects.map(app => app.permitNumber ? app.permitNumber.toUpperCase() : null)).sort());
+
+      // (re)apply filtering
+      this.internalApplyFilters(false);
+    }
+  }
+
+  public handleRadioChange(value) {
+    this.radioSel = value;
+  }
+
+  public applyFilters() {
+    // clear previous search terms
+    if (this.radioSel === 'Mine Name' ) {
+      this._permitFilter = null;
+    } else {
+      this._mineFilter = null;
+    }
+    this.mineFilter = this._mineFilter;
+    this.permitFilter = this._permitFilter;
+    this.ranSearch = true;
+    this.internalApplyFilters(true);
+  }
+
+  private internalApplyFilters(doSave: boolean) {
+    this.projects.forEach(mine => mine.isMatch = this.showThisMine(mine));
+    let projCount = this.projects.filter(mine => mine.isMatch);
+    this.resultsCount = projCount.length;
+    this.updateMatching.emit();
+
+    if (doSave) {
+      this.saveFilters();
+    }
+  }
+
+  private showThisMine(item: Project): boolean {
+    let retVal = true;
+
+    const mineFilter = this.mineFilter && this.mineFilter.trim(); // returns null or empty
+    retVal = retVal && (
```

Comment:
`retVal` will always be `true` here, so I don't think it needs to be checked. 

```retVal = (......```

Reply:
correct this retVal wasn't needed for this part of the logic, it is needed for the permitFilter or we override the value and mine search would return everything